storage/storage.go

```diff
@@ -120,13 +120,26 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error
 		opts = append(opts, internaloption.WithDefaultEndpoint("https://storage.googleapis.com/storage/v1/"))
 		opts = append(opts, internaloption.WithDefaultMTLSEndpoint("https://storage.mtls.googleapis.com/storage/v1/"))
 	} else {
-		scheme = "http"
-		readHost = host
```

Comment:
Is readHost no longer set here? Why not?

Reply:
As discussed offline, we are removing these temporary variables as the scheme and readHost are set later anyway from the endpoint provided by `htransport`, which should be valid